action.yml

```diff
@@ -0,0 +1,103 @@
+---
+name: 'Docker-build-release'
+description: 'Build Docker image and push to GHCR'
+inputs:
+  organization:
+    description: 'Organizational host for the image. Defaults to the organization of the calling repository.'
+  metadata-file:
+    description: 'Metadata YAML file containing information'
+    default: metadata.yaml
+  image-name-key-path:
+    description: 'YAML path to key for image name'
+    default: '.image_name'
+  github-token:
```

Comment:
**Question**: Is there risk of this input being exposed? See these [logs](https://productionresultssa11.blob.core.windows.net/actions-results/55879840-423f-4364-b7d5-96aa8f1c4f79/workflow-job-run-bc02be18-d32e-56fd-9beb-cc49dd5488df/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-08-16T23%3A30%3A57Z&sig=r8t2b3ZOfELmpBDJH6rMNNG8KCTS6CBskRU9NG0rfIU%3D&ske=2024-08-17T10%3A08%3A08Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-08-16T22%3A08%3A08Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-05-04&sp=r&spr=https&sr=b&st=2024-08-16T23%3A20%3A52Z&sv=2024-05-04) for example, from a debug run of the workflow:

```
024-08-16T23:20:37.6745682Z ##[debug]..Evaluating Index:
2024-08-16T23:20:37.6747790Z ##[debug]....Evaluating inputs:
2024-08-16T23:20:37.6749529Z ##[debug]....=> Object
2024-08-16T23:20:37.6763825Z ##[debug]....Evaluating String:
2024-08-16T23:20:37.6764438Z ##[debug]....=> 'force_frontend'
2024-08-16T23:20:37.6767457Z ##[debug]..=> 'true'
2024-08-16T23:20:37.6769399Z ##[debug]..Evaluating Index:
2024-08-16T23:20:37.6769941Z ##[debug]....Evaluating inputs:
2024-08-16T23:20:37.6770594Z ##[debug]....=> Object
2024-08-16T23:20:37.6771044Z ##[debug]....Evaluating String:
2024-08-16T23:20:37.6771541Z ##[debug]....=> 'force_backend'
2024-08-16T23:20:37.6772264Z ##[debug]..=> 'true'
2024-08-16T23:20:37.6772741Z ##[debug]..Evaluating Index:
2024-08-16T23:20:37.6773306Z ##[debug]....Evaluating inputs:
2024-08-16T23:20:37.6773783Z ##[debug]....=> Object
2024-08-16T23:20:37.6774239Z ##[debug]....Evaluating String:
2024-08-16T23:20:37.6774816Z ##[debug]....=> 'force_meta'
2024-08-16T23:20:37.6775282Z ##[debug]..=> 'false'
```

Reply:
So the non-debug build logs do redact the inputs:

```
Run uclahs-cds/tool-Docker-action@dev
  with:
    organization: uclahs-cds
    metadata-file: metadata.yaml
    image-name-key-path: .image_name
    github-token: ***
    context: .
```
I can't positively state that they therefore handle all cases, but at least _something_ about this situation was predicted.

Reply:
I'm wondering whether that's only the case when `github.token` is used. If someone put in a different token what happens?

Reply:
If it's passed as a Secret, I believe it will be properly redacted: https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#redacting-secrets-from-workflow-run-logs

If it is provided in plain-text there will be no protection.

Reply:
Good find!

Reply:
> An attacker can exfiltrate any stolen secrets or other data from the runner. To help prevent accidental secret disclosure, GitHub Actions [automatically redact secrets printed to the log](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#accessing-your-secrets), but this is not a true security boundary because secrets can be intentionally sent to the log. For example, obfuscated secrets can be exfiltrated using `echo ${SOME_SECRET:0:4}; echo ${SOME_SECRET:4:200};`. In addition, since the attacker may run arbitrary commands, they could use HTTP requests to send secrets or other repository data to an external server.